monix-tail/shared/src/test/scala/monix/tail/IterantRepeatSuite.scala

```diff
@@ -113,6 +114,18 @@ object IterantRepeatSuite extends BaseTestSuite {
     assertEquals(source4.repeat.toListL.value(), List.empty[Int])
   }
 
+  test("Iterant.repeat discards Scopes properly") { implicit s =>
+    val acquired = Atomic(0)
+    val sum = Iterant
+      .resource(Coeval(acquired.incrementAndGet()))(_ => Coeval(acquired.decrement()))
```

Comment:
This actually won't discriminate vs old behavior where finalizers are executed at the end. You need to check that list is `List(1, 1, 1, 1, 1, 1...)` at least

Reply:
It does discriminate, because with the old behavior you get `1, 2, 3, ... 10` as elements, so you get the total sum to be `55` instead of `10`.

I can test with a list of course, instead of a sum.

Reply:
Right, disregard that 😓